pandas/tests/indexes/test_interval.py

```diff
@@ -491,6 +649,207 @@ def testcontains(self):
         assert not i.contains(20)
         assert not i.contains(-20)
 
+        # Why are there two tests called test contains? this should be cleared up a little, no?
+
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_interval_covers_interval_updated_behavior(self):
+
+        # class Interval:
+        #     def covers(self, other: Interval) -> bool
+
+        assert     Interval(1, 3).covers(Interval(1.5, 2.5))
+        assert     Interval(1, 3).covers(Interval(1, 2))
+        assert     Interval(1, 3).covers(Interval(2, 3))
+        assert not Interval(1, 3).covers(Interval(0.5, 2.5))
+        assert not Interval(1, 3).covers(Interval(1.5, 3.5))
+
+        assert     Interval(1, 3, closed='right').covers(Interval(1, 3, closed='right'))
+        assert not Interval(1, 3, closed='right').covers(Interval(1, 3, closed='left'))
+        assert not Interval(1, 3, closed='right').covers(Interval(1, 3, closed='both'))
+
+        assert not Interval(1, 3, closed='left').covers(Interval(1, 3, closed='right'))
+        assert     Interval(1, 3, closed='left').covers(Interval(1, 3, closed='left'))
+        assert not Interval(1, 3, closed='left').covers(Interval(1, 3, closed='both'))
+
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='right'))
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='left'))
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='both'))
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_interval_covers_intervalIndex_updated_behavior(self):
+
+        # class Interval:
+        #     def covers(self, other: IntervalIndex) -> IntegerArray1D
```

Comment:
We will have to discuss the exact semantics of such a new method.  
Personally I would expect a boolean array as result here. 

But maybe we can discuss that in a separate PR / issue, to keep it here about the indexing routines? (already complex enough :-))